projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/vdk_kerberos_client.py

```diff
@@ -0,0 +1,166 @@
+        else:
+            kdc_req_body["etype"] = override_etype
+
+        pa_data_1 = {}
+        pa_data_1["padata-type"] = int(PADATA_TYPE("PA-PAC-REQUEST"))
+        pa_data_1["padata-value"] = PA_PAC_REQUEST({"include-pac": True}).dump()
+
+        kdc_req = {}
+        kdc_req["pvno"] = krb5_pvno
+        kdc_req["msg-type"] = MESSAGE_TYPE.KRB_AS_REQ.value
+        kdc_req["padata"] = [pa_data_1]
+        kdc_req["req-body"] = KDC_REQ_BODY(kdc_req_body)
+
+        req = AS_REQ(kdc_req)
+
+        logger.debug("Sending initial TGT to %s" % self.ksoc.get_addr_str())
+        rep = await self.ksoc.sendrecv(req.dump())
+
+        if rep.name != "KRB_ERROR":
+            # user can do kerberos auth without preauthentication!
+            self.kerberos_TGT = rep.native
+
+            # if we want to roast the asrep (tgt rep) part then we dont even have the proper keys to decrypt
+            # so we just return, the asrep can be extracted from this object anyhow
+            if decrypt_tgt == False:
+                return
+
+            self.kerberos_cipher_type = kdc_req_body["etype"][
+                0
+            ]  # patched: retrieve first encryption type supported
+            self.kerberos_cipher = _enctype_table[self.kerberos_cipher_type]
+            self.kerberos_key = Key(
+                self.kerberos_cipher.enctype,
+                self.usercreds.get_key_for_enctype(
+                    EncryptionType(self.kerberos_cipher_type)
+                ),
+            )
+
+        else:
+            if (
+                rep.native["error-code"]
+                != KerberosErrorCode.KDC_ERR_PREAUTH_REQUIRED.value
+            ):
+                raise KerberosError(rep)
+            rep = rep.native
+            logger.debug("Got reply from server, asikg to provide auth data")
+            supported_encryption_method = self.select_preferred_encryption_method(rep)
+
+            rep = await self.do_preauth(supported_encryption_method)
+            logger.debug("Got valid TGT response from server")
+            rep = rep.native
+            self.kerberos_TGT = rep
+
+        if self.usercreds.certificate is not None:
+            (
+                self.kerberos_TGT_encpart,
+                self.kerberos_session_key,
+                self.kerberos_cipher,
+            ) = self.decrypt_asrep_cert(rep)
+            self.kerberos_cipher_type = supported_encryption_method.value
+
+            self.ccache.add_tgt(
+                self.kerberos_TGT, self.kerberos_TGT_encpart, override_pp=True
+            )
+            logger.debug("Got valid TGT")
+            return
+
+        else:
+            cipherText = self.kerberos_TGT["enc-part"][
+                "cipher"
+            ]  # patched: correct ciphertext lookup
+            temp = self.kerberos_cipher.decrypt(self.kerberos_key, 3, cipherText)
+
+            try:
+                self.kerberos_TGT_encpart = EncASRepPart.load(temp).native
+            except Exception as e:
```

Comment:
nit: Maybe log the error message, as well, or remove `e` altogether?

Reply:
same one, not marked with patched, would like to keep it same way for easier back-contribution